test/integration/performance/connection_http_transport_benchmark_test.py

```diff
@@ -0,0 +1,264 @@
+- by default will re-run multiple times without resetting fully -> this means for import
+ that the data becomes much greater over time
+
+need to prepare data in efficient/universal-way as input -> faker?, cheats with export?
+worthwhile with import to create large file or just string together x rows, y times.
+
+do test against
+- csv -> direct DB call
+- file
+- iterable
+- pandas
+- parquet
+- polars
+
+discuss results with Torsten before too much development to see if right direction
+"""
+
+"""
+Hmm, based on an AI response to the question, we have
+Column sizes:
+  SALES_ID            :   18 bytes
+  SALES_TIMESTAMP     :   26 bytes
+  PRICE               :   11 bytes
+  CUSTOMER_NAME       :  200 bytes
+
+CSV formatting overhead:
+  quotes              :   10 bytes
+  delimiters          :    4 bytes
+  newline             :    2 bytes
+
+Total row size: 271 bytes
+
+Rows per GB: ~4 million
+
+Batch processing test:
+Time to generate 100,000 rows in batches: 1.34 seconds
+Estimated time for 4M rows with batch processing: 50.77 seconds (0.01 hours)
+"""
+
+EXPORT_FROM_TABLE = "SALES"
+IMPORT_INTO_TABLE = "TMP_SALES_COPY"
+INITIAL_SIZE = 1_000
+FINAL_SIZE = 4_000_000
+
+
+def create_empty_table(connection: ExaConnection, table_name: str):
+    ddl = cleandoc(
+        f"""
+        CREATE OR REPLACE TABLE {table_name} (
+            SALES_ID                DECIMAL(18,0) IDENTITY NOT NULL PRIMARY KEY,
+            SALES_TIMESTAMP         TIMESTAMP,
+            PRICE                   DECIMAL(9,2),
+            CUSTOMER_NAME           VARCHAR(200)
+          );
+        """
+    )
+    connection.execute(ddl)
+    connection.commit()
+
+
+@pytest.fixture(scope="session")
+def session_connection(connection_factory):
+    con = connection_factory()
+    yield con
+    con.close()
+
+
+@pytest.fixture(scope="session")
+def tmp_source_directory(tmp_path_factory) -> Path:
+    return tmp_path_factory.mktemp("benchmark")
+
+
+@pytest.fixture(scope="session")
+def create_empty_sales_table(session_connection: ExaConnection):
+    create_empty_table(connection=session_connection, table_name=EXPORT_FROM_TABLE)
+    yield EXPORT_FROM_TABLE
+
+    ddl = f"DROP TABLE IF EXISTS {EXPORT_FROM_TABLE};"
+    session_connection.execute(ddl)
+    session_connection.commit()
+
+
+@pytest.fixture(scope="session")
+def create_initial_sales_data(
+    create_empty_sales_table, session_connection: ExaConnection
+):
+    query = f"""
+        INSERT INTO {EXPORT_FROM_TABLE} (SALES_TIMESTAMP, PRICE, CUSTOMER_NAME)
+        SELECT ADD_SECONDS(TIMESTAMP'2024-01-01 00:00:00', FLOOR(RANDOM() * 365 * 24 * 60 * 60)),
+        RANDOM(1, 125),
+        SUBSTRING(
+        LPAD(TO_CHAR(FLOOR(RANDOM() * 1000000)), 6, '0'),
+        1,
+        FLOOR(RANDOM() * 3) + 5)
+        FROM dual
+        CONNECT BY level <= {INITIAL_SIZE}
+    """
+    session_connection.execute(query)
+    session_connection.commit()
+
+
+# def fill_table(connection: ExaConnection):
```

Comment:
could use merge/union to double, but how does this benefit us for the import testing? for sure in the export one this is good.

Reply:
Yes, this is more for export.